dev/bots/prepare_package.dart

```diff
@@ -61,131 +66,314 @@ class ArchiveCreator {
 
   final Directory flutterRoot;
   final Directory tempDir;
-  final File outputFile;
-  final ProcessRunner _runner;
+  final Directory outputDirectory;
+  final bool _subprocessOutput;
+  final ProcessStarter _starter;
   String flutter;
   final String git = Platform.isWindows ? 'git.bat' : 'git';
```

Comment:
package:process would also hide from you that the executables on Windows have different endings.

Reply:
How does it do that?  It could be git.bat or git.exe, and I happen to know what we're using, which is why I set it here.

Reply:
package:process resolves `git` to an executable just like the Windows shell does: It looks at the list of extensions in the `PATHEXT` env variable and checks one by one if a file with that ending and the given name exists. If yes, that is the executable used.

I find that approach more future proved as there is no guarantee that the executable will continue to be named `git.bat`. It could very well be `git.exe` in a future release.